url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/data/gathering/dataSource/ReutersNews.java

```diff
@@ -0,0 +1,188 @@
+package darwInvest.data.gathering.dataSource;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.Scanner;
+
+import darwInvest.data.NewsEvent;
+import darwInvest.data.Ticker;
+import darwInvest.data.utility.*;
+
+/**
+ * The GoogleFinanceAuto generates and reads in CSVs for the stocks in
+ * GoogleFinanceAuto/tickers.txt.
+ * 
+ * @author Andrew Perrault, Kevin Dolan
+ */
+
+public class ReutersNews extends DataSource {
+
+	private String source;
+
+	public ReutersNews(Serializer serializer, Date startdate, Date enddate) {
+		super("ReutersNews", serializer, startdate, enddate);
+	}
+
+	private String webize(String name) {
+		char[] arr = name.toCharArray();
+		for(int i = 0; i < arr.length; i++) {
+			if (arr[i] == ' ') {
+				arr[i] = '+';
+			}
+		}
+		return new String(arr);
+	}
+
+	private Date dconv(String date) {
+		GregorianCalendar cal = new GregorianCalendar();
+		GoogleFinanceManual csvreader = new GoogleFinanceManual(new DefaultSerializer());
+		int day = 0, month = 0, year = 0;
+		String mon = null;
+		if (date.length() == 15) {
+			day = Integer.parseInt(date.substring(9, 10));
+			mon = date.substring(5, 8);
+			year = Integer.valueOf((date.substring(11)));
+		}
+		if (date.length() == 16) {
+			day = Integer.parseInt(date.substring(9, 11));
+			mon = date.substring(5, 8);
+			year = Integer.valueOf(date.substring(12));
+		}
+		month = csvreader.GoogletoMonth(mon);
+		cal.set(year, month, day);
+		return cal.getTime();
+	}
+
+	private Boolean inRange(Date startdate, Date enddate, Date date) {
+		if(enddate.after(date) && startdate.before(date)) {
+			return true;
+		}
+		else return false;
+	}
+
+	@Override
+	public void gatherData() {
+		//no of stories to read in, for time's sake.
+		int nostories = 15;
+		File target = new File("StockData/" + source);
+		target.mkdir();
+		File listoftickers = new File("RawData/" + super.getName() + "/tickers.txt");
+		Scanner s = null;
+		try {
+			s = new Scanner(new BufferedReader(new FileReader(listoftickers)));
+			s.useDelimiter("[\n\r]+");
+			while(s.hasNext()) {
+				String currentstock = s.next();
+				Ticker ticker = new Ticker(currentstock);
+				String companyname = ticker.lookupCompanyname();
+				String reuters = "http://www.reuters.com/search?blob=%22" + webize(companyname) +  "%22";
+				readIn(ticker, reuters, nostories);
+				addTicker(ticker);
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			if (s != null) {
+				s.close();
+			}
+		}
+	}
+
+	private void readIn(Ticker ticker, String reuters, int nostories) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.